library/system/src/lib/yast2/system_service.rb

```diff
@@ -239,27 +246,43 @@ def current_start_mode
 
     # Sets start mode to the underlying system
     def save_start_mode
-      return unless changed_value?(:start_mode)
-      case new_value_for(:start_mode)
-      when :on_boot
-        service.enable
-        socket.disable
-      when :on_demand
-        service.disable
-        socket.enable
-      when :manual
-        service.disable
-        socket.disable
-      end
+      return unless changes[:start_mode]
+      result =
+        case changes[:start_mode]
+        when :on_boot
+          service.enable && socket.disable
+        when :on_demand
+          service.disable && socket.enable
+        when :manual
+          service.disable && socket.disable
+        end
+      register_error(:start_mode) unless result
```

Comment:
to be honest I do not like much this error attribute way. It is a bit like C style. Can we add least add save! that will raise well defined exception when it failed?

Reply:
Well, it is more like Rails style. But yes, we can improve it later, I will add a new task to the PBI.